### ISSUE TITLE
Live Preview Not Working

### ISSUE THREAD

**Issue by levinelson** [craftcms/digital-products#84]
### Description

Upgraded to Craft 4 a month or so ago, and just realize Live Preview does not seem to work when editing Digital Products. Live Preview *is* working when editing other vanilla Craft entries.

### Steps to reproduce

1. Open a digital product entry page in CP
2. Click "Live Preview"
3. In the console, I see a `Failed to load resource: the server responded with a status of 500 ()` for something like `https://path-to-entry?token=-h7B4f57cIfo_IHiWnP6l54NGNYsept8`.
4. Going to that URL above also says `Bad Request Post request required`.

### Additional info

- Craft version: Craft Pro 4.3.7
- PHP version: 8.0.27
- Database driver & version: MySQL 8.0.32
- Plugins & versions: Commerce 4.2.5.1, Digital Products 3.2.0

**Comment by levinelson** [comment]
Contents of log that might be relevant:
```2023-02-09 01:47:47 [web.ERROR] [TypeError] Cannot assign null to property craft\base\Element::$enabled of type bool {"trace":["#0 /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/digital-products/src/controllers/ProductsController.php(244): craft\\digitalproducts\\controllers\\ProductsController->_buildProductFromPost()","#1 [internal function]: craft\\digitalproducts\\controllers\\ProductsController->actionPreviewProduct()","#2 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)","#3 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams(Array)","#4 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/Module.php(552): yii\\base\\Controller->runAction('preview-product', Array)","#5 /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/cms/src/web/Application.php(301): yii\\base\\Module->runAction('digital-product...', Array)","#6 /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/cms/src/controllers/LivePreviewController.php(117): craft\\web\\Application->runAction('digital-product...')","#7 [internal function]: craft\\controllers\\LivePreviewController->actionPreview('digital-product...', 1)","#8 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)","#9 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams(Array)","#10 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/Module.php(552): yii\\base\\Controller->runAction('preview', Array)","#11 /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/cms/src/web/Application.php(301): yii\\base\\Module->runAction('live-preview/pr...', Array)","#12 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/web/Application.php(103): craft\\web\\Application->runAction('live-preview/pr...', Array)","#13 /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/cms/src/web/Application.php(286): yii\\web\\Application->handleRequest(Object(craft\\web\\Request))","#14 /srv/users/serverpilot/apps/happy-toolbox/vendor/yiisoft/yii2/base/Application.php(384): craft\\web\\Application->handleRequest(Object(craft\\web\\Request))","#15 /srv/users/serverpilot/apps/happy-toolbox/public/index.php(21): yii\\base\\Application->run()","#16 {main}"],"memory":6929160,"exception":"[object] (TypeError(code: 0): Cannot assign null to property craft\\base\\Element::$enabled of type bool at /srv/users/serverpilot/apps/happy-toolbox/vendor/craftcms/digital-products/src/controllers/ProductsController.php:509)"} 
2023-02-09 01:47:47 [web.WARNING] [application] Request context: {"userId":1,"sessionId":"5800ee6d203682847512a33cff001653","vars":{"_GET":{"p":"pack/everyday-people","token":"••••••••••••••••••••••••••••••••"},"_POST":{"title":"Everyday People","sku":"everyday-people-2","price":"55","typeId":"1","productId":"76024","siteId":"1"},"_COOKIE":{"1031b8c41dfff97a311a7ac99863bdc5_identity":"0789aa8e563323d8b33189a8f992f8d5f0b4c09c8d86a14f8956992e18214ecaa:2:{i:0;s:41:\"1031b8c41dfff97a311a7ac99863bdc5_identity\";i:1;s:162:\"[1,\"[\\\"D9OIjX3VR1ybw5dHAWHSaSJVa1xshPQ1FD_9bjg-lrw6PCgbQb8eBBV1D7JxCzM3zrZW_uIzpbh6u5TF_Ub8rJCcf6PC-aVnbhWG\\\",null,\\\"ec8c936136150844c6d12856ff49c7cc\\\"]\",1209600]\";}","e349f3cd050b66ac7b4f9b12d0345804_commerce_cart":"662632abcef9bd5b5a7a89decceac6e095b058ab5c22e001ca8dee785433f1f0a:2:{i:0;s:46:\"e349f3cd050b66ac7b4f9b12d0345804_commerce_cart\";i:1;s:32:\"171ca2a8c88b4e5bf08cc6c5a826f094\";}","1031b8c41dfff97a311a7ac99863bdc5_username":"fd033e0011712d13597470a44f1288eb0f9358433abcc59e35392f3358a2f2f6a:2:{i:0;s:41:\"1031b8c41dfff97a311a7ac99863bdc5_username\";i:1;s:25:\"admin@thehappytoolbox.com\";}","CRAFT_CSRF_TOKEN":"••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••","CraftSessionId":"5800ee6d203682847512a33cff001653"},"_FILES":[],"_SERVER":{"TMP":"/srv/users/serverpilot/tmp/happy-toolbox","TEMP":"/srv/users/serverpilot/tmp/happy-toolbox","TMPDIR":"/srv/users/serverpilot/tmp/happy-toolbox","PATH":"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin","USER":"serverpilot","HOME":"/srv/users/serverpilot","SCRIPT_NAME":"/index.php","REQUEST_URI":"/pack/everyday-people?token=rFfkR1bn-dOzLhNaY7TM_GpcRYU0ZgWm","QUERY_STRING":"p=pack/everyday-people&token=rFfkR1bn-dOzLhNaY7TM_GpcRYU0ZgWm","REQUEST_METHOD":"POST","SERVER_PROTOCOL":"HTTP/1.0","GATEWAY_INTERFACE":"CGI/1.1","REDIRECT_QUERY_STRING":"p=pack/everyday-people&token=rFfkR1bn-dOzLhNaY7TM_GpcRYU0ZgWm","REDIRECT_URL":"/pack/everyday-people","REMOTE_PORT":"31404","SCRIPT_FILENAME":"/srv/users/serverpilot/apps/happy-toolbox/public/index.php","SERVER_ADMIN":"webmaster@","CONTEXT_DOCUMENT_ROOT":"/srv/users/serverpilot/apps/happy-toolbox/public","CONTEXT_PREFIX":"","REQUEST_SCHEME":"https","DOCUMENT_ROOT":"/srv/users/serverpilot/apps/happy-toolbox/public","REMOTE_ADDR":"104.28.97.19","SERVER_PORT":"443","SERVER_ADDR":"127.0.0.1","SERVER_NAME":"thehappytoolbox.com","SERVER_SOFTWARE":"Apache/2.4.55 (Unix) OpenSSL/1.1.1j","SERVER_SIGNATURE":"","LD_LIBRARY_PATH":"/opt/sp/apache/lib:/opt/sp/apache/lib","HTTP_COOKIE":"1031b8c41dfff97a311a7ac99863bdc5_identity=0789aa8e563323d8b33189a8f992f8d5f0b4c09c8d86a14f8956992e18214ecaa%3A2%3A%7Bi%3A0%3Bs%3A41%3A%221031b8c41dfff97a311a7ac99863bdc5_identity%22%3Bi%3A1%3Bs%3A162%3A%22%5B1%2C%22%5B%5C%22D9OIjX3VR1ybw5dHAWHSaSJVa1xshPQ1FD_9bjg-lrw6PCgbQb8eBBV1D7JxCzM3zrZW_uIzpbh6u5TF_Ub8rJCcf6PC-aVnbhWG%5C%22%2Cnull%2C%5C%22ec8c936136150844c6d12856ff49c7cc%5C%22%5D%22%2C1209600%5D%22%3B%7D; e349f3cd050b66ac7b4f9b12d0345804_commerce_cart=662632abcef9bd5b5a7a89decceac6e095b058ab5c22e001ca8dee785433f1f0a%3A2%3A%7Bi%3A0%3Bs%3A46%3A%22e349f3cd050b66ac7b4f9b12d0345804_commerce_cart%22%3Bi%3A1%3Bs%3A32%3A%22171ca2a8c88b4e5bf08cc6c5a826f094%22%3B%7D; 1031b8c41dfff97a311a7ac99863bdc5_username=fd033e0011712d13597470a44f1288eb0f9358433abcc59e35392f3358a2f2f6a%3A2%3A%7Bi%3A0%3Bs%3A41%3A%221031b8c41dfff97a311a7ac99863bdc5_username%22%3Bi%3A1%3Bs%3A25%3A%22admin%40thehappytoolbox.com%22%3B%7D; CRAFT_CSRF_TOKEN=08b5916efc6e43ff435199e1b43f02e28b8d32f6b689545baadfe7a166447ac7a%3A2%3A%7Bi%3A0%3Bs%3A16%3A%22CRAFT_CSRF_TOKEN%22%3Bi%3A1%3Bs%3A147%3A%22Jcp__bHzDdQo4B5Lufu42b_n3iLw6Y9aqcLT3gE7%7C50491de2a0aabe0ee092ff2f3c832b4f6a7d7aaa5c9f4790afe4c61681c65959Jcp__bHzDdQo4B5Lufu42b_n3iLw6Y9aqcLT3gE7%7C1%22%3B%7D; CraftSessionId=5800ee6d203682847512a33cff001653","HTTP_REFERER":"https://thehappytoolbox.com/admin/digital-products/products/pack/76024","HTTP_USER_AGENT":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15","HTTP_ORIGIN":"https://thehappytoolbox.com","HTTP_ACCEPT_LANGUAGE":"en-US,en;q=0.9","HTTP_ACCEPT_ENCODING":"gzip, deflate, br","HTTP_X_CRAFT_TOKEN":"••••••••••••••••••••••••••••••••","HTTP_ACCEPT":"*/*","CONTENT_TYPE":"application/x-www-form-urlencoded; charset=UTF-8","CONTENT_LENGTH":"86","HTTP_CONNECTION":"close","HTTP_X_FORWARDED_PROTO":"https","HTTP_X_FORWARDED_SSL":"on","HTTP_X_FORWARDED_FOR":"104.28.97.19","HTTP_HOST":"thehappytoolbox.com","SCRIPT_URI":"https://thehappytoolbox.com/pack/everyday-people","SCRIPT_URL":"/pack/everyday-people","HTTPS":"on","REDIRECT_STATUS":"200","REDIRECT_SCRIPT_URI":"https://thehappytoolbox.com/pack/everyday-people","REDIRECT_SCRIPT_URL":"/pack/everyday-people","REDIRECT_HTTPS":"on","FCGI_ROLE":"RESPONDER","PHP_SELF":"/index.php","REQUEST_TIME_FLOAT":1675907266.911806,"REQUEST_TIME":1675907266,"argv":["p=pack/everyday-people&token=rFfkR1bn-dOzLhNaY7TM_GpcRYU0ZgWm"],"argc":1,"ENVIRONMENT":"production","SECURITY_KEY":"••••••••••••••••••••••••••••••••","DB_DRIVER":"mysql","DB_SERVER":"localhost","DB_USER":"7e28f2a65505","DB_PASSWORD":"••••••••••••••••••••","DB_DATABASE":"happy-toolbox","DB_SCHEMA":"public","DB_TABLE_PREFIX":"","DB_PORT":"3306","DEFAULT_SITE_URL":"https://thehappytoolbox.com","STRIPE_PUBLISHABLE_KEY":"••••••••••••••••••••••••••••••••••••••••••","STRIPE_SECRET_KEY":"••••••••••••••••••••••••••••••••••••••••••","PAYPAL_CLIENT_ID":"AVzgkG29j3_WWuhGjhnpTRh36ZxN2NVm4_c2IZrBLRh6zh8wK1DUzjMrvhAiTNXaXs4qngjXQ6HelAZg","PAYPAL_SECRET":"••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••","PAYPAL_TEST_MODE":"false","GOOGLE_PROJECT_ID":"bright-feat-248901","GOOGLE_BASE_URL":"https://storage.googleapis.com/happy-toolbox","GOOGLE_APPLICATION_CREDENTIALS":"/srv/users/serverpilot/apps/happy-toolbox/config/google-cloud.json","SENDGRID_API_KEY":"•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••","MAILCHIMP_API_KEY":"•••••••••••••••••••••••••••••••••••••","MAILCHIMP_AUDIENCE_ID":"3e9fd7a417","KNOCK_KNOCK_PASSWORD":"•••••••••••••••••","RECAPTCHA_SITE_KEY":"••••••••••••••••••••••••••••••••••••••••","RECAPTCHA_SECRET_KEY":"••••••••••••••••••••••••••••••••••••••••","MIXPANEL_TOKEN":"••••••••••••••••••••••••••••••••"},"_SESSION":{"bd62416aa8538ede709019a5e113eea5__flash":[],"1031b8c41dfff97a311a7ac99863bdc5__token":"••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••","1031b8c41dfff97a311a7ac99863bdc5__id":1,"__authKey":"••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••","1031b8c41dfff97a311a7ac99863bdc5__expire":1677116866,"__duration":1209600,"bd62416aa8538ede709019a5e113eea5__auth_access":["saveAssets:2","saveAssets:1","saveAssets:5"]}}} 
```

**Comment by lukeholder** [comment]
This was fixed in 3.2.1 Thanks.
